### PR TITLE
Search Item Api  Implementation

### PR DESCRIPTION
**Pull request by kevinle623**
Due to difficulties, I accidentally auto merged the previous pull request. I have reverted but the previous pull request for this implementation is now closed. From this, I am opening another a new one. Since then, I have add string docs to each of the functions in the api as well as commenting certain lines of codes to be more clear. Apologies again for the spam.

### CODE REVIEW

**Review by Ljime — APPROVED** [review]

**Review by chinchinjim — DISMISSED** [review]
looks good 👍

File: src/main/java/Item.java [hunk 1]
```diff
@@ -91,8 +91,11 @@ public void displayItemInConsole(int ranking){
         System.out.println("------------------------------------------");
     }
 
+    /** Returns custom url link with respective keyword search and marketplace
```
Comment (PyroPoro) [inline src/main/java/Item.java:94]: is this the docstring for updatePrice? if so, it doesn't seem to be correct. The function doesnt return anything. It only mutates the item's price attribute

Comment (kevinle623) [reply src/main/java/Item.java]: I corrected the docstring for updatePrice. Thanks for the review.

**Review by AlexChenl3 — APPROVED** [review]
everything looks good kevin

**Review by AlexChenl3 — APPROVED** [review]

**Review by AlexChenl3 — APPROVED** [review]

**Review by Ljime — APPROVED** [review]

**Review by PyroPoro — APPROVED** [review]

**Pull request merged**